api/apiclient.go

```diff
@@ -665,7 +665,12 @@ func startDialWebsocket(ctx context.Context, try *parallel.Try, addr, path strin
 		Total: opts.Timeout,
 		Delay: opts.RetryDelay,
 	}
-	if openAttempt.Min == 0 && openAttempt.Delay > 0 {
+	// Workaround for systems with poor clock resolution on which it's possible
+	// to try to connect 4 times in 0 seconds.
+	if openAttempt.Total == 0 && openAttempt.Delay == 0 {
+		openAttempt.Delay = 1
```

Comment:
A delay of 1ns won't help matters too much, I think.
I'd think I'd prefer to fix TestDialAPIMultipleError than at
this logic to the production code - if we want production
code to be able to use the zero DialOpts, I think we should
decide on some semantics, document them, implement them,
and fix the tests that rely on the current semantics.

I'd suggest something like this for the semantics:

	// Timeout is the amount of time to wait contacting
	// a controller.	If this is zero, the Dial may block forever.
	Timeout time.Duration

	// RetryDelay is the amount of time to wait between
	// unsuccessful connection attempts. If this is
	// zero, no retries will be made.
	RetryDelay time.Duration